tonic/models/vic/grid_params.py

```diff
@@ -917,6 +1150,34 @@ def grid_params(soil_dict, target_grid, snow_dict, veglib_dict, veg_dict,
                 for y, x in pyzip(ymask, xmask):
                     out_dict[var][:, :, y, x] = fill_val
 
+            elif mydict[var].ndim == 4:
+                j = mydict[var].shape[1]
+                k = mydict[var].shape[2]
+                l = mydict[var].shape[3]
+                out_dict[var] = np.ma.zeros((j, k, l, ysize, xsize),
+                                            dtype=dtype)
+                for jj in pyrange(j):
+                    for kk in pyrange(k):
+                        for ll in pyrange(l):
+                            out_dict[var][jj, kk, ll, yi, xi] = mydict[var][:, jj, kk, ll]
```

Comment:
I'm almost entirely sure this can be done without any for loops.  Numpy is great when you can leverage its vectorization. Were you having problems with the brodcasting here?